packages/flutter/test/widgets/sliver_main_axis_group_test.dart

```diff
@@ -855,6 +855,26 @@ void main() {
 
     expect(tester.getTopLeft(find.byKey(firstTextFieldKey)), const Offset(0, 60));
   });
+
+  testWidgets('SliverMainAxisGroup multiple PinnedHeaderSliver children', (WidgetTester tester) async {
```

Comment:
Can you add tests for each permutation of scroll direction and reverse?

Reply:
+1 to this, it'd be nice to double-check that each possible configuration works correctly.

Reply:
I encountered some issues when testing in reverse. I have submitted an issue for this:#159787. I believe this problem should not be fixed in this PR, so I have only added a TODO.

Reply:
We should fix that issue then before landing this - a la embrace the yak shave!
I think I have a fix for it, just need to add a couple tests!

Reply:
I meant to include a reference for yak shaving: https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#lazy-programming

🙂

Reply:
Update: Still working on a fix. I found that once I fixed localToGlobal, WidgetTester.getRect broke. 🙃 

So I need to do a bit more digging. Sharing the draft for folks that are curious, https://github.com/flutter/flutter/pull/160203. I think I know where to look next, just might not find time until next week. :)